From 7adefadcd79a33ba625a948c42745a7ff51325f6 Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Wed, 31 May 2017 15:45:26 +0200
Subject: [PATCH 168/169] tc: matchall: Print skip flags when dumping a filter

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1435624
Upstream status: iproute2.git commit afdc1fed243f5

commit afdc1fed243f5499a53f5fda202031cf4c4d4044
Author: Or Gerlitz <ogerlitz@mellanox.com>
Date:   Thu Feb 9 15:10:14 2017 +0200

    tc: matchall: Print skip flags when dumping a filter

    Print the skip flags when we dump a filter.

    Signed-off-by: Or Gerlitz <ogerlitz@mellanox.com>
    Acked by: Yotam Gigi <yotamg@mellanox.com>
    Reviewed-by: Simon Horman <simon.horman@netronome.com>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
---
 tc/f_matchall.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/tc/f_matchall.c b/tc/f_matchall.c
index 04e524e3..ac486308 100644
--- a/tc/f_matchall.c
+++ b/tc/f_matchall.c
@@ -130,6 +130,15 @@ static int matchall_print_opt(struct filter_util *qu, FILE *f,
 			sprint_tc_classid(rta_getattr_u32(tb[TCA_MATCHALL_CLASSID]), b1));
 	}
 
+	if (tb[TCA_MATCHALL_FLAGS]) {
+		__u32 flags = rta_getattr_u32(tb[TCA_MATCHALL_FLAGS]);
+
+		if (flags & TCA_CLS_FLAGS_SKIP_HW)
+			fprintf(f, "\n  skip_hw");
+		if (flags & TCA_CLS_FLAGS_SKIP_SW)
+			fprintf(f, "\n  skip_sw");
+	}
+
 	if (tb[TCA_MATCHALL_ACT])
 		tc_print_action(f, tb[TCA_MATCHALL_ACT]);
 
-- 
2.13.0

